httpbakery/client.go

```diff
@@ -261,18 +261,25 @@ func (c *Client) doWithBody(req *http.Request, body io.ReadSeeker) (*http.Respon
 	if err != nil {
 		return nil, errgo.Mask(err, errgo.Any)
 	}
+	var cookiePath string
 	cookieURL := req.URL
 	if path := resp.Info.MacaroonPath; path != "" {
 		relURL, err := parseURLPath(path)
 		if err != nil {
 			logger.Warningf("ignoring invalid path in discharge-required response: %v", err)
 		} else {
-			cookieURL = req.URL.ResolveReference(relURL)
+			cookieURL := req.URL.ResolveReference(relURL)
+			cookiePath = cookieURL.Path
 		}
 	}
-	if err := SetCookie(c.Jar, cookieURL, macaroons); err != nil {
+	if len(cookiePath) > 1 && strings.HasSuffix(cookiePath, "/") {
```

Comment:
if think:

   cookiePath = strings.TrimSuffix(cookiePath, "/")

should be sufficient here.